src/flow-worker.c

```diff
@@ -66,6 +68,13 @@ typedef struct FlowWorkerThreadData_ {
  */
 static inline void FlowUpdate(Packet *p)
 {
+
+    int state = SC_ATOMIC_GET(p->flow->flow_state);
```

Comment:
I would prefer this to move into FlowHandlePacketUpdate, as that is more 'core flow engine'